### PR TITLE
[Contrastive Trainer] Check for duplicate projectors

### LINKED ISSUE #1887: Double gradient calculation for the projectors in Contrastive Trainer
https://github.com/keras-team/keras-cv/blob/b3798d8140605b651ac46f40fe7072ea08175720/keras_cv/training/contrastive/contrastive_trainer.py#L233-L238

https://github.com/keras-team/keras-cv/blob/b3798d8140605b651ac46f40fe7072ea08175720/keras_cv/training/contrastive/contrastive_trainer.py#L240-L247

In the above two code snippets we assume that both the projectors are distinct. What if we provide only one projector which is then replicated to fill a tuple as shown below 👇 

https://github.com/keras-team/keras-cv/blob/b3798d8140605b651ac46f40fe7072ea08175720/keras_cv/training/contrastive/contrastive_trainer.py#L112-L114

In that case the gradients of the projector is calculated twice and also the projector is updated twice.

If I have read this code snippet differently and am mistaken please feel free to close this issue.

CC: @ID6109

### PR DESCRIPTION
**Pull request by ariG23498**
Fixes #1887 

CC: @ianstenbit

### CODE REVIEW

File: keras_cv/training/contrastive/contrastive_trainer.py [hunk 1]
```diff
@@ -109,6 +109,8 @@ def __init__(
             augmenter if type(augmenter) is tuple else (augmenter, augmenter)
         )
         self.encoder = encoder
+        # Check to see if the projector is being shared or are distinct.
+        self._is_shared_projector = False if type(projector) is tuple else True
```
Comment (ianstenbit) [inline keras_cv/training/contrastive/contrastive_trainer.py:113]: How about `not isinstance(projector, tuple)`?

File: keras_cv/training/contrastive/contrastive_trainer.py [hunk 2]
```diff
@@ -230,19 +232,23 @@ def train_step(self, data):
                 regularization_losses=self.encoder.losses,
             )
 
+        # If the projector is shared, then take the trainable weights of just
+        # one of the projector in the tuple. If not, use both the projectors.
```
Comment (ianstenbit) [inline keras_cv/training/contrastive/contrastive_trainer.py:236]: nit: "one of the projectors"

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thanks! Looks good after a few minor changes

Also -- contrastive tests are not run on CI, so please run `pytest --run_extra_large keras_cv/training/contrastive`

**Comment by ariG23498** [comment]
The tests pass locally!

**Review by ianstenbit — APPROVED** [review]
Thanks for the fix!
